Makefile

```diff
@@ -208,6 +208,44 @@ tidy-check: gen_cmake ## Run clang-tidy
 			 \nFor detail information about the checks, please refer to https://clang.llvm.org/extra/clang-tidy/checks/list.html" \
 			 ; exit 1)
 
+# TO detect unused functions in cppcheck, we must have only 1 job
+NUM_JOBS:=8
+ifeq (${DETECT_UNUSED_FUNC},TRUE)
+	NUM_JOBS:=1
+endif
+
+.PHONY: cppcheck
+cppcheck: gen_cmake ## Run Cppcheck static analysis: `make cppcheck REL_PATH=<relative path-to-folder/file,optional> DETECT_UNUSED_FUNC=<TRUE|FALSE,optional> `. REL_PATH (optional) is a relative path to CONCORD_BFT_TARGET_SOURCE_PATH. In that case, Cppcheck analyzes only a single file or all files under the given folder. Wildcards are supported, for example: 'REL_PATH=bftengine/src/preprocessor/GlobalData.*' includes both hpp/cpp files. If REL_PATH is not defined, Cppcheck analyzes all source code under CONCORD_BFT_TARGET_SOURCE_PATH, excluding files and folders defined under option -i (ignore). It is recommended to enable DETECT_UNUSED_FUNC only when REL_PATH yields multiple translation units. If DETECT_UNUSED_FUNC is enabled, the number of parallel jobs is decreased to 1. It is useful to have multiple jobs (8 by default) only when there are multiple translation units. For better analysis includes.txt and suppressions.txt files are under CONCORD_BFT_TARGET_SOURCE_PATH/.cppcheck.
+	@if [ ! -f ${REL_PATH} ] && [ ! -d ${REL_PATH} ]; then echo 'Error: REL_PATH=${REL_PATH} is not a valid file/path!'; exit 1; fi;
+	docker run ${CONCORD_BFT_USER_GROUP} ${BASIC_RUN_PARAMS} \
+		${CONCORD_BFT_CONTAINER_SHELL} -c \
+		"set -eo pipefail && \
+		cd ${CONCORD_BFT_BUILD_DIR} && \
+		rm -rf cppcheck/reports/${REL_PATH}/ && \
+		mkdir -p cppcheck/reports/${REL_PATH}/ && \
+		cppcheck  --template='cppcheck1'  --inconclusive --enable=all --platform=unix64 --std=c++17 -j ${NUM_JOBS} --force --quiet \
```

Comment:
Please export to a bash script and add comments explaining what you were doing and why.  This is not an easy read.

Reply:
I will.